scripts/chk_shellscripts/chk_shellscripts.sh

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env bash
+
+set -e
+
+REPO_ROOT="$(dirname "$0")"/../..
+REPO_ROOT=$(realpath "${REPO_ROOT}")
+BASELINE_FILENAME="baseline.checksums"
+BASELINE_FILE="${REPO_ROOT}/scripts/chk_shellscripts/${BASELINE_FILENAME}"
+
+for arg in "$@"; do
+  case "$arg" in
+  --create-baseline) CREATE_BASELINE=1 ;;
+  *)
+    echo "Usage:"
+    echo "  $(basename "${0}") [--create-baseline]"
+    echo "  --create-baseline    create a baseline of current shell scripts. all errors"
+    echo "                       that where found during this step, will be ignored until a "
+    echo "                       script was changed. once it was change, all errors need to be fixed."
+    exit 0
+    ;;
+  esac
+done
+
+create_baseline() {
+  echo -n "" >"${BASELINE_FILE}"
+  for FILE in "${FILES[@]}"; do
+    HASH=$(shasum -U -a 256 "${REPO_ROOT}/${FILE}" | awk '{ print $1 }')
+    echo "SHA256 (${FILE}) = ${HASH}" >>"${BASELINE_FILE}"
+  done
+
+  exit 0
+}
+
+check_baseline() {
+  while IFS='' read -r line; do
```

Comment:
We should stick as much as possible to C++ coding style, so `do` should go to next line
Same for `then` in case of `if`

Reply:
Hmm.. I'm not sure.. this style is quite often used within our shell scripts..

Reply:
I think the fact that you have to add a `;` suggests that `do` should not be put on the same line.

Reply:
hmm.. `do` can only be in the same line, if `;` was used.. however, the majority of shell scripts are not using this, so I can change that..